### PR TITLE
Add option to ignore hidden files on import

### LINKED ISSUE #110: Ignore OS-specific hidden files
This issue was automatically migrated from Google Code.
_Original author: Joris.Dobbelaer@gmail.com (January 13, 2012 19:58:52)_
_Original issue: https://github.com/google-code-export/beets/issues/293_

### PR DESCRIPTION
**Pull request by jackwilsdon**
This fixes #110 by adding a top-level `ignore_hidden` option to the configuration.

### CODE REVIEW

File: beets/util/hidden.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    (FILE_ATTRIBUTE_HIDDEN).
+    """
+    # FILE_ATTRIBUTE_HIDDEN = 2 (0x2) from GetFileAttributes documentation.
+    hidden_mask = 2
+
+    # Retrieve the attributes for the file.
+    attrs = ctypes.windll.kernel32.GetFileAttributesW(path)
+
+    # Ensure we have valid attribues and compare them against the mask.
+    return attrs >= 0 and bool(attrs & hidden_mask)
```
Comment (sampsyo) [inline beets/util/hidden.py]: I don't think the `bool` cast is necessary here—Python will do this automatically (i.e., zero is falsey, nonzero is truthy).

**Comment by sampsyo** [comment]
This looks awesome; thank you! You should get a special award for fixing such an old issue.

Please merge at will (with a changelog entry). :tada:

File: beets/util/hidden.py (hunk 1 above)
Comment (jackwilsdon) [inline beets/util/hidden.py]: Ah yes, it was an accidental remnant from when I didn't have `attrs >= 0`.

**Comment by jrobeson** [comment]
what about adding the directory lost+found as seen on ext* filesystems on linux?

**Comment by jrobeson** [comment]
oh, ignore that, it's not actually hidden, but it should still be ignored

**Comment by sampsyo** [comment]
@jrobeson, maybe it should go into the default `ignore` config option?

**Comment by jrobeson** [comment]
done in #1995

**Pull request merged**